### PR TITLE
#166789955 Users should be able to view and edit their profiles

### PR DESCRIPTION
**Pull request by IsaiahRn**
#### What does this PR do?

- Users should be able to view and edit their profiles

#### Description of Task to be completed

- Add view and edit functionality

- Display input error validation

- Add tests

#### How should this be manually tested?

1. Run the following in your terminal
```
git clone https://github.com/andela/ah-92explorers-frontend.git
cd ah-92explorers-frontend
git checkout ch-setup-e2e-tests-166789946
npm install
npm run dev
```

#### Any background context you want to provide?

- N/A

#### What are the relevant pivotal tracker stories?
[#166789955](https://www.pivotaltracker.com/story/show/166789955)
#### Screenshots (if appropriate)

- View Updated User Profile
Screenshot 2019-08-25 23 35 35

- Edit User Profile
Screenshot 2019-08-22 22 12 26

- Input validation alerts
Screenshot 2019-08-25 23 36 27

Screenshot 2019-08-25 23 36 57

#### Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I don't have more than 3 major commits on this PR

### CODE REVIEW

File: src/redux/reducers/profile.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  profile: null,
+  loading: true,
+  error: null,
+};
+
+export default function (state = initialState, action) {
+  const { type, payload } = action;
+
+  switch (type) {
+    case GET_PROFILE:
```
Comment (kodek-sleuth) [inline src/redux/reducers/profile.js:18]: By doing this Case statement without expecting to return something is not inadequate. Its like creating an if staement like this;
```
if (GET_PROFILE){

} 
Refactor this in a way that if you expect a case, it should return something like you did in the case block after 
```

**Review by kodek-sleuth — CHANGES_REQUESTED** [review]

File: src/redux/reducers/profile.js (hunk 1 above)
Comment (IsaiahRn) [reply src/redux/reducers/profile.js:18]: > By doing this Case statement without expecting to return something is not inadequate. Its like creating an if staement like this;
> 
> ```
> if (GET_PROFILE){
> 
> } 
> Refactor this in a way that if you expect a case, it should return something like you did in the case block after 
> ```

Thanks! I am going to work on that.

**Review by elemanhillary-zz — CHANGES_REQUESTED** [review]
@IsaiahRn I have seen that you are doing a shallow test on the components, for example, me as a `TTL` i want to be able to see that you have simulated button clicks in your test for the component to prove that the `buttons` actually react to click `events` and one more thing test the whole component to test for some `HTML elements` existence comparing them with want you as a developer expects.

**Review by Hervera — CHANGES_REQUESTED** [review]
You have too many commits

**Review by Hervera — CHANGES_REQUESTED** [review]
I am not able to make a command of `npm install` after switching to this branch. There is a bug of `SyntaxError: Unexpected end of JSON input while parsing near '...`

**Comment by IsaiahRn** [comment]
> I am not able to make a command of `npm install` after switching to this branch. There is a bug of `SyntaxError: Unexpected end of JSON input while parsing near '...`

@Hervera , I think that this is a problem related to your npm, you can try to look into that?

**Review by niyoceles — DISMISSED** [review]
Good work

**Review by Hervera — CHANGES_REQUESTED** [review]
Test your Navbar. It has `F` point on `coverage`

**Comment by niyoceles** [comment]
> I am not able to make a command of `npm install` after switching to this branch. There is a bug of `SyntaxError: Unexpected end of JSON input while parsing near '...`

@Hervera delete `node_modules` folder and `package-lock.json`, then run `npm install'

**Review by Hervera — DISMISSED** [review]
Ready to be merged

**Review by niyoceles — DISMISSED** [review]
Good job

**Review by kodek-sleuth — DISMISSED** [review]

File: src/components/Layout/navBar.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+          <img src={logo} alt="" className="logo dash-logo" />
+          <ul>
+            <li><a href="./userDashboard.html">Dashboard</a></li>
+            <li><a href="./stats.html">Stats</a></li>
+            <li><a href="./profile.html">Profile</a></li>
+          </ul>
+          <div className="restOfNav">
+            <img src={search} alt="" className="navIcons navIconColor" />
+            <img src={notification} alt="" className="navIcons navIconColor" />
+            <img src={manIcon} alt="" className="navIcons manIcon" />
```
Comment (IsaiahRn) [inline src/components/Layout/navBar.js]: Use the actual profile picture

File: src/redux/actions/actionCreators/profile.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    window.location.reload(true);
+    dispatch(performAction(PROFILE_ERROR, error.response));
+  }
+};
+
+export const updateProfile = (profileData, history) => async (dispatch) => {
+  checkToken();
+  try {
+    const formData = new FormData();
+    formData.append('firstName', profileData.firstName);
```
Comment (IsaiahRn) [inline src/redux/actions/actionCreators/profile.js]: Remove code repetition

**Review by niyoceles — APPROVED** [review]

**Review by kodek-sleuth — APPROVED** [review]

**Review by elemanhillary-zz — APPROVED** [review]
ready for merge

**Pull request merged**